pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -129,11 +129,43 @@ func (i *KumaInjector) needInject(pod *kube_core.Pod, ns *kube_core.Namespace) (
 		log.V(1).Info("pod fulfills exception requirements")
 		return false, nil
 	}
-	enabled, exist, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaSidecarInjectionAnnotation)
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name == util.KumaSidecarContainerName {
+			log.V(1).Info("pod already has Kuma sidecar")
+			return false, nil
+		}
+	}
+
+	enabled, exist, err := metadata.Annotations(pod.Labels).GetEnabled(metadata.KumaSidecarInjectionAnnotation)
```

Comment:
Probably worth noting since the pod usually overrides the namespace due to specificity - if you label namespace, pod annotation will not override. Should pod annotation get be up here to override namespace label? Seems like a priority inversion as currently written.

Reply:
yes, I agree, now it's like `pod.Label > ns.Label > pod.Annotation > ns.Annotation` but probably it should be `pod.Label >  pod.Annotation > ns.Label  > ns.Annotation` since we support both labels and annotations for some period of time

Reply:
Thanks @parkanzky , that's a good point about order